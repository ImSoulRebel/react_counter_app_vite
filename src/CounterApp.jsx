import { useState } from "react";
import PropTypes from "prop-types";

export function CounterApp({ value }) {
  const [counter, setCounter] = useState(value);
  const addNumber = () => {
    setCounter(counter + 1);
    // setCounter((c)=> c+1);
  };
  const substractNumber = () => {
    setCounter(counter - 1);
    // setCounter((c)=> c+1);
  };
  const resetNumber = () => {
    setCounter(value);
    // setCounter((c)=> c+1);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={addNumber}>+1</button>
      <button onClick={substractNumber}>-1</button>
      <button aria-label="button-reset" onClick={resetNumber}>
        reset
      </button>
    </>
  );
}

CounterApp.PropTypes = {
  value: PropTypes.number.isRequired, //Genera error que bloquea codigo
};
