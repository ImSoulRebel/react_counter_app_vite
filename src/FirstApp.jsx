/* eslint-disable react/prop-types */
import PropTypes from 'prop-types';

// Variables y funciones fuera de Functional component para evitar reasignamiento de memoria
// const newMessage = 'Bruce!!';

export const FirstApp = ({title, subtitle, name}) =>{
  return (
    // Fragment, para retornar varios elementos .jsx dentro de un nodo padre
    <> 
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  )
}

// Obliga a seguir tipados en React+JScript
FirstApp.PropTypes = {
    title: PropTypes.string.isRequired, //Genera error que bloquea codigo
    subtitle: PropTypes.number,
    name: PropTypes.string,
}

FirstApp.defaultProps = {
  name: 'nombre por defecto',
  title: 'No hay titulo',
  subtitle: 0,
}
