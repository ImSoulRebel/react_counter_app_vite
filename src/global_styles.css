export html,body{
    background-color:  #21232A;
    color:white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 7px;
}

button{
    padding: 15px;
    font-size: 1.3rem;
    margin: 5px;
}