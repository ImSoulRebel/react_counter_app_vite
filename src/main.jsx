import React from 'react';
import ReactDOM from 'react-dom/client';
// import {FirstApp} from './FirstApp';
// import {HelloWorldApp} from './HelloWorldApp'
import './global_styles.css';
import {CounterApp} from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={23}/>
        {/* < FirstApp title="Hello, i'm Vegeta"/> */}
    </React.StrictMode>
);