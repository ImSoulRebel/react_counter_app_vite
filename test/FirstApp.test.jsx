/**
 * @jest-environment jsdom
 */
import {  render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


/* eslint-disable no-undef */
describe('Test on <FirstApp.test/>', () => { 


    test('Match with snapshot', () => { 
        const title = 'I am  Goku';
        const {container} = render(<FirstApp title={ title }/>);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar el titulo en un h1', () => { 
        const title = 'I am  Goku';
        const {container, getByTestId} = render(<FirstApp title={ title }/>);
        // expect( getByText(title) ).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);
        expect(getByTestId('test-title').innerHTML).toContain(title);
     });

     test('debe mostrar screen el titulo en un h1', () => { 
        const title = 'I am  Goku';
         render(<FirstApp title={ title }/>);     
        expect(screen.getByRole('heading', {level: 1}) .innerHTML).toContain(title);
     });
 });