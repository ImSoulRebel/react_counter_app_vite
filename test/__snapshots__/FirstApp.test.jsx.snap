// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <FirstApp/> Match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    I am  Goku
  </h1>
  <p>
    0
  </p>
  <p>
    nombre por defecto
  </p>
</div>
`;
